spring-pulsar/src/main/java/org/springframework/pulsar/core/reactive/ReactivePulsarSenderTemplate.java

```diff
@@ -79,22 +91,44 @@ public void setSchema(Schema<T> schema) {
 	private Mono<MessageId> doSend(String topic, T message,
 			MessageSpecBuilderCustomizer<T> messageSpecBuilderCustomizer, MessageRouter messageRouter,
 			ReactiveMessageSenderBuilderCustomizer<T> customizer) {
-		final String topicName = ReactiveMessageSenderUtils.resolveTopicName(topic, this.reactiveMessageSenderFactory);
-		this.logger.trace(() -> String.format("Sending reative msg to '%s' topic", topicName));
+		return doSendMany(topic, Mono.just(message), messageSpecBuilderCustomizer, messageRouter, customizer).single();
+	}
 
-		final ReactiveMessageSender<T> sender = createMessageSender(topic, message, messageRouter, customizer);
+	private Flux<MessageId> doSendMany(String topic, Publisher<T> messages,
+			MessageSpecBuilderCustomizer<T> messageSpecBuilderCustomizer, MessageRouter messageRouter,
+			ReactiveMessageSenderBuilderCustomizer<T> customizer) {
+		final String topicName = ReactiveMessageSenderUtils.resolveTopicName(topic, this.reactiveMessageSenderFactory);
+		this.logger.trace(() -> String.format("Sending reactive msg to '%s' topic", topicName));
+
+		if (this.schema != null) {
+			ReactiveMessageSender<T> sender = this.reactiveMessageSenderFactory.createReactiveMessageSender(topic,
```

Comment:
I would be in favor of just calling into `createMessageSender` as it hides a bit of ternary customizer below - it does the correct thing w/ the schema.

Reply:
done